### PR TITLE
v1.13 Backports 2023-12-07

### PR DESCRIPTION
**Pull request by nbusseneau**
* [x] #29623 (@pchaigno)
 * [x] #29653 (@brb)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 29623 29653
```

### CODE REVIEW

**Comment by nbusseneau** [comment]
@pchaigno Conflicts were hit while backporting your commits, please check backporter's notes for details :)

**Review by pchaigno — APPROVED** [review]

**Review by ti-mo — APPROVED** [review]

**Review by brb — APPROVED** [review]
Thanks!

**Comment by brb** [comment]
The ci-ipsec-upgrade seems to be failing due to the same reason - https://github.com/cilium/cilium/issues/29351. cc @qmonnet

**Comment by qmonnet** [comment]
> The ci-ipsec-upgrade seems to be failing due to the same reason - #29351. cc @qmonnet

Yes for kernels 5.4 and 5.10.

Not for kernel 6.1. The failure is the same as observed for 6.1 on https://github.com/cilium/cilium/pull/29703, but it's new to me.

**Comment by nbusseneau** [comment]
/test-1.26-net-next

**Comment by nbusseneau** [comment]
@qmonnet So my assumption is that we're safe to ignore failures in `ci-ipsec-upgrade` because known failures + it's not marked as required, correct?

**Comment by giorio94** [comment]
Bumped the Cilium CLI to the latest version (v0.15.17), as `conn-disrupt-dispatch-interval` was introduced in v0.15.8.

**Comment by giorio94** [comment]
/test-backport-1.13

**Comment by giorio94** [comment]
Rebased onto the tip of v1.13 to pick https://github.com/cilium/cilium/commit/286d113f4c419e40be60bc5f9bd4b73ec356ec15 and fix Travis.

**Comment by giorio94** [comment]
/test-backport-1.13

**Comment by giorio94** [comment]
IPsec failures are either https://github.com/cilium/cilium/issues/29805 or https://github.com/cilium/cilium/issues/29351

**Comment by giorio94** [comment]
/test-1.22-4.19

**Comment by ldelossa** [comment]
@giorio94 just to be clear, this is ready to merge despite failures?

**Comment by giorio94** [comment]
> @giorio94 just to be clear, this is ready to merge despite failures?

https://github.com/cilium/cilium/issues/29351 is a known flake which is being looked into, and the check is not required, so I'd say we are good for that. 
https://github.com/cilium/cilium/issues/29805 seems introduced by this PR (which adds the CI check, no code changes). I'd like to get @brb opinion here whether we should stand by for the moment.

**Comment by brb** [comment]
Let's maybe hold off merging this PR until I have investigated the 6.1 failures (will do it until EOD tomorrow).

File: .github/workflows/tests-ipsec-upgrade.yaml [hunk 1]
```diff
@@ -98,6 +98,15 @@ jobs:
...
             ipv6: 'false' # due to https://github.com/cilium/cilium/issues/23461
 
+          - config: '6.1'
+            # renovate: datasource=docker depName=quay.io/lvh-images/kind
+            kernel: '6.1-20231026.065108'
+            kube-proxy: 'iptables'
+            kpr: 'disabled'
+            tunnel: 'vxlan'
+            encryption: 'ipsec'
+            endpoint-routes: 'false'
```
Comment (brb) [inline .github/workflows/tests-ipsec-upgrade.yaml:108]: @nbusseneau Could you append the following to the commit which added this config?
```
# https://github.com/cilium/cilium/issues/29805#issuecomment-1855600171
ipv6: 'false'
```

**Comment by brb** [comment]
/test-backport-1.13

**Comment by brb** [comment]
/test-backport-1.13

**Pull request merged**